sklearn/cluster/_agglomerative.py

```diff
@@ -743,6 +743,11 @@ class AgglomerativeClustering(ClusterMixin, BaseEstimator):
 
         .. versionadded:: 0.21
 
+    return_distance : bool, default=False
```

Comment:
+1 for compute_distances

Also the docstring should indicate that this is ignored when `distance_threshold` is used.